### ISSUE TITLE
Track Conversion/Transaction Chain

### ISSUE THREAD

**Issue by WhoSoup** [pegnet/pegnet#284]
PegNet needs to sync with the Conversion/Transaction chain and parse them, maintaining a state of balances.

* The chain's entries are in the FAT format, which includes things like replay protection and the entry signing standard
* Conversions and Transactions use the same base format and only differentiate in parameters. This is described in #283 

Requirements for implementation are:
* Transactions can be done in the same block
* Conversions are done at the end of the **next** block using the values of that block's winner
* This data needs to be accessible by the command line, the api, and a database so it should be modularized/queryable/persistable
* Implement the balances without using floats since we're dealing with financial data.  @Emyrk has already done work in this area 

This work is redundant with a fatd implementation, so there is an opportunity here to create a ** module** that can be imported by both fatd and pegnet in order to save a whole lot of development time.

**Comment by Emyrk** [comment]
I'm unsure if a module will be compatible with bot fat and other projects since I assume FAT has a specific sqllite format?

**Comment by Emyrk** [comment]
On a call discussing pM2, miners will have to use a FATd as a data source to determine the value of PEG before the PEG value is on exchange. This is because to implement conversions+rewards in pegnet to determine things like market cap would be a large duplication of FATd.

**Comment by WhoSoup** [comment]
Progress report on implementing the Grading part of tracking balances:

I managed to successfully get fatd to recognize the OPR chain, parse it, and keep the rates in the sql database. This unfortunately comes with a set of problems that adam and i will have to work out:

* Syncing takes a long time (30+ minutes) which isn't relayed well to the end user
* Implementation paradigms required by FAT

The work-in-progress code is available at https://github.com/WhoSoup/fatd/tree/pegnet-opr2 but in order to run it, you'll need to respect the relative gomod entry and initialize a blank go mod file in the various module repos. This situation will be improved once the grading module PR is finished.

The next step will be getting sam's transaction chain to be recognized, at which point we can merge the two branches and have the opr chain insert the coinbase into the transaction chain's database.
